docs/_source/docs/stack_config.rst

```diff
@@ -248,17 +288,56 @@ referring to is in a different AWS account or region.
 
 stack_tags
 ~~~~~~~~~~
+* Resolvable: Yes
+* Can be inherited from StackGroup: Yes
+* Inheritance strategy: Overrides parent if set
 
 A dictionary of `CloudFormation Tags`_ to be applied to the Stack.
 
 stack_timeout
 ~~~~~~~~~~~~~
+* Resolvable: No
+* Can be inherited from StackGroup: Yes
+* Inheritance strategy: Overrides parent if set
 
 A timeout in minutes before considering the Stack deployment as failed. After
 the specified timeout, the Stack will be rolled back. Specifiyng zero, as well
 as ommiting the field, will result in no timeout. Supports only positive
 integer value.
 
+.. _project_dependency_config:
+
+is_project_dependency
+~~~~~~~~~~~~~~~~~~~~~
+* Resolvable: No
+* Can be inherited from StackGroup: Yes
+* Inheritance strategy: Overrides parent if set
+
+A project dependency stack is intended to be a stack that is utilized by other stacks in the project,
+or even all stacks. This is useful for defining stacks to support the project, such as a template bucket,
+SNS notification topics, and/or an IAM service role.
+
+The biggest difference between a "normal" stack config and a stack marked as a project dependency is
+that stacks marked with ``is_project_dependency: True`` will ignore all dependencies of their own.
+Project dependencies are not allowed to have dependencies. Furthermore, the
+**the stack_output resolver will always resolve to nothing on project dependency stacks.**
+
+For example, this allows you to safely set the ``template_bucket_name`` on the top-level StackGroup
+config as the output from a project_dependency stack. This means that every **other** stack not marked
+as a project dependency will use that template_bucket_name, while the template bucket stack will
+have NO template bucket name (i.e. its template will not be uploaded anywhere).
+
+Another example would be defining the role_arn on the top-level stack config with the ``!stack_output``
+of a project dependency stack that outputs the role arn. Every **other** stack will use this stack
+as a CloudFormation service role, but project dependency stacks will have no service role.
```

Comment:
It's difficult to get the jist of all of this without some concrete template examples.  would be good to add examples to cover some of the most common use cases.

Reply:
I have been thinking that the "sample architectures" page could use an overhaul. The changes from these PRs seem like they really could improve that. I'll work this weekend, I think, on adding to that page's docs with concrete examples of the sort of setup that is now supported. I can link to there from here.

Reply:
I added a detailed example project (something we never really had and probably needed anyway) in the "Architecting Sceptre" page: https://github.com/Sceptre/sceptre/pull/1157/files#diff-498f17109556dd683849428994e46e8731d43ab7fff3688fd78d5fd4591003c2

This should make things a lot clearer, I think.